crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -43,154 +42,464 @@ mod validate_op_tests;
+        let current_validation_dependencies = current_validation_dependencies.clone();
+        let incoming_dht_ops_sender = incoming_dht_ops_sender.clone();
+        async move {
+            let state = {
+                let mut deps = current_validation_dependencies.lock();
+                if let Some(state) = deps.get(&hash) {
+                    state.set_source(CascadeSource::Local);
+                }
 
-    // trigger other workflows
-    trigger_app_validation.trigger(&"sys_validation_workflow");
+                deps.get(&hash).cloned()
+            };
+
+            if let Some(state) = state {
+                let op_type = state.required_by_op_type();
+                let record = state.as_record().cloned();
+
+                if let (Some(op_type), Some(record)) = (op_type, record) {
+                    let result = match op_type {
+                        DhtOpType::StoreRecord => {
+                            incoming_dht_ops_sender
+                                .send_store_record(
+                                    record,
+                                )
+                                .await
+                        }
+                        DhtOpType::StoreEntry => {
+                            incoming_dht_ops_sender
+                                .send_store_entry(
+                                    record,
+                                )
+                                .await
+                        }
+                        DhtOpType::RegisterAgentActivity => {
+                            incoming_dht_ops_sender
+                                .send_register_agent_activity(
+                                    record,
+                                )
+                                .await
+                        }
+                        DhtOpType::RegisterAddLink => {
+                            incoming_dht_ops_sender
+                                .send_register_add_link(
+                                    record,
+                                )
+                                .await
+                        }
+                        _ => {
+                            tracing::warn!(op_type = ?op_type, "Unexpected op type for network fetched dependency");
+                            Ok(())
+                        }
+                    };
 
-    Ok(complete)
+                    if let Err(e) = result {
+                        tracing::error!(error = ?e, "Error sending network fetched dependency to incoming dht ops");
+                    }
+                } else {
+                    tracing::warn!("Missing op type or record for network fetched dependency");
+                }
+            } else {
+                tracing::warn!("Missing state for network fetched dependency");
+            }
+        }
+        .boxed()
+    }))
+    .await;
+
+    if num_fetched < outcome_summary.missing {
+        Ok(WorkComplete::Incomplete(Some(
```

Comment:
I'd like to refactor this so it uses a suspending loop but I felt that was an extra change too much on this PR which is already too big :)